src/System.Management.Automation/utils/PathUtils.cs

```diff
@@ -515,5 +542,104 @@ internal static Encoding Convert(Cmdlet cmdlet, string encoding)
 
             return null;
         }
+
+        // We need a way to provide an encoding and also
+        // a way to notify that the user wants "byte" encoding
+        // which is just a stream
+        internal class ByteEncoding : System.Text.Encoding
```

Comment:
Should this class and the EncodingConversion.byteEncoding moved here as a public field?

In other words, is the problem you're solving with this custom instance be something third-party cmdlets will need to solve as well?

Reply:
this is really a dodge for backward compatibility. `-Encoding Byte` should have never existed, but it does and I don't want to have this be a difference between 6 and 5.x. This _byte_ encoding is really needed for the *-content cmdlets where the parameter is used as a flag indicating that the user is looking for a stream of bytes, I would rather not make this public

Reply:
Since we're already making this big breaking change, I think we should fix this by not having `ByteEncoding` as `Byte` is an artificial "encoding".  The `*-Content` cmdlets should have a different parameter to output or receive a byte stream.

Reply:
It would more clear to users to have -Byte or -Binary parameter.
@mklement0 @rkeithhill What do you think?

Reply:
I kind of prefer `-AsByteArray` myself.

cc @HemantMahawar @joeyaiello

Reply:
Purely from a usability perspective, I don't have a problem with `-Encoding Byte`, because even though it is technically not a _text_ encoding, the `-Encoding` parameter can be conceptualized as being about how to _interpret_ the data on reading and writing, and `-Encoding Byte` (a byte stream) is one way of (non-)interpreting .

This keeps all considerations regarding encoding/non-encoding focused on a single parameter, which simplifies things (it doesn't introduce another parameter that you have to learn about and remember), both in terms of use and documentation.

Reply:
How can we interpret `Set-Content -Encoding 'Ru-ru' -Byte`?